applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_custom_scipy_solver.py

```diff
@@ -0,0 +1,147 @@
+import KratosMultiphysics
+
+# Import applications
+import KratosMultiphysics.StructuralMechanicsApplication as StructuralMechanicsApplication
+
+# Import base class file
+from KratosMultiphysics.StructuralMechanicsApplication.structural_mechanics_solver import MechanicalSolver
+
+from KratosMultiphysics import python_linear_solver_factory as linear_solver_factory
+
+# Import scipy modules
+import KratosMultiphysics.scipy_conversion_tools
+
+def CreateSolver(main_model_part, custom_settings):
+    return CustomScipySolver(main_model_part, custom_settings)
+
+class CustomScipySolver(MechanicalSolver):
+    """The structural mechanics custom scipy solver.
+
+    This class creates the mechanical solvers to provide mass and stiffness matrices as scipy matrices.
+    
+    Derived class must override the function SolveSolutionStep. In there the Mass and Stiffness matrices
+    can be obtained as scipy matrices.
+
+    See structural_mechanics_solver.py for more information.
+    """
+    def __init__(self, main_model_part, custom_settings):
+        # Construct the base solver.
+        super(CustomScipySolver, self).__init__(main_model_part, custom_settings)
+        KratosMultiphysics.Logger.PrintInfo("::[CustomScipySolver]:: ", "Construction finished")
+        self.space = KratosMultiphysics.UblasSparseSpace()
+
+    @classmethod
+    def GetDefaultSettings(cls):
+        this_defaults = KratosMultiphysics.Parameters("""{
+            "scheme_type"         : "dynamic"
+        }""")
+        this_defaults.AddMissingParameters(super(CustomScipySolver, cls).GetDefaultSettings())
+        return this_defaults
+
+    #### Private functions ####
+    def _create_solution_scheme(self):
+        """Create the scheme for the scipy solver.
+
+        The scheme determines the mass and stiffness matrices 
+        """
+        scheme_type = self.settings["scheme_type"].GetString()
+        if scheme_type == "dynamic":
+            solution_scheme = StructuralMechanicsApplication.EigensolverDynamicScheme()
+        else: # here e.g. a stability scheme could be added
+            err_msg =  "The requested scheme type \"" + scheme_type + "\" is not available!\n"
+            err_msg += "Available options are: \"dynamic\""
+            raise Exception(err_msg)
+
+        return solution_scheme
+
+    def _create_linear_solver(self):
+        ''' Linear solver will not be used. But eventually the solution stratgey calls the solver's clear function. 
+        To avoid crashing linear solver is provided here'''
+        return linear_solver_factory.CreateFastestAvailableDirectLinearSolver()
+
+    def _create_mechanical_solution_strategy(self):
+        if self.settings["block_builder"].GetBool() == True:
+            warn_msg = "In case an eigenvalue problem is computed an elimantion builder shall be used to ensure boundary conditions are applied correctly!"
+            KratosMultiphysics.Logger.PrintWarning("CustomScipySolver", warn_msg)
```

Comment:
Would an error work in this case? Or do you need the possibility to use a 'block_builder'

Reply:
Depends on what the user wants to do. I did it this way to be more general. In case user wants to have full matrix. If elimination builder only is enough error message would be better.

Reply:
Ok for me.

Reply:
I don't understand why one cannot use the block-builder here

Reply:
it is not strictly right that you cannot use it. the problem is that the eigenvalue associated to the diagonal which corresponds to the dirichlet modes(the "1s" in the diagonal is only controlled by the  value of the number you set in there. Depending on the chosen value it may interfere with the other eigenvalues